### DISCUSSION TITLE
Check if field is required using yup and RHF

### DISCUSSION THREAD

**Discussion by McGern** [react-hook-form/react-hook-form#5258] [Q&A]
Hi,

Been using react hook form and yup in our react project and it is going pretty well, and I am looking for a way to utilise the validation rules in the yup schema to set a indicator on the field about whether it is required or not.

It seems trivial to say just add a star * to the field label, but when you have a conditionally mandatory field, or if the mandatory requirement on the fields is coming from rules set on the server (as we have in my case, when on one contract, a mobile number might be mandatory, but on another contract it isn't - but they are both using the same form, just different rules), it would easier to manage this in one place.

Have been searching for a while in RHF github and stack overflow, but can't find any mention of this, and the closest I could find was this post on the formik issue log https://github.com/formium/formik/issues/712, which talks about the same thing (but with yup and formik).

So I am wondering how do you (can you?) expose the rules that state whether a field is required or not?

Thanks,
Jarrod

**Comment by fredrivett** [comment]
Based on @d-asensio's [answer for Formik](https://github.com/jaredpalmer/formik/issues/712#issuecomment-452749640), this is working for me for `react-hook-form`:

```javascript
const schema = yup
  .object({
    title: yup.string().required("Title is required"),
  })
  .required();

const isRequiredField = name =>
  !!schema
    .describe()
    .fields[name].tests.find(test => test.name === "required");

console.log(isRequiredField("title")); // —> true
```

**Comment by leapful** [comment]
@McGern @fredrivett : Actually, you can use something like a context to keep checking on required state of each field, but then you need to use the latest version of yup (>= 1.0.0) in order to perform conditional checking by using method "describe".

Example:

```
const buildSchema = (context) => {
  return yup.object({
    colors: yup.string().required("Required"),
    name: yup.string().when("colors", {
      is: "red",
      then: () => {
        context["name"].required = true;

        return yup.string().required();
      },
      otherwise: () => {
        context["name"].required = false;

        return yup.string();
      }
    })
  });
};
```

Working demo: https://codesandbox.io/s/react-hook-form-yup-resolver-forked-9i0j0u?file=/src/App.js

**Reply by McGern** [reply]
That looks interesting, thanks!
